piptools/utils.py

```diff
@@ -725,7 +725,11 @@ def _invert_negative_bool_options_in_config(
     for key, value in config.items():
         # Transform config key to its equivalent in the CLI
         long_option = _convert_to_long_option(key)
-        new_key = cli_opts[long_option].name if long_option in cli_opts else key
+        new_key = (
+            cli_opts[long_option].name
+            if long_option in cli_opts
+            else key.replace("no_", "")
+        )
```

Comment:
```suggestion
        new_key = cli_opts[long_option].name if long_option in cli_opts else key
        if new_key.startswith('no_'):
            new_key = new_key[3:]
```

This is more explicit. This issue with `key.replace("no_", "")` is that it can potentially break options like `--no-code-no-errors`.